src/navigators/configurable-tabs/MessageTileStack.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useRoute } from '@react-navigation/native';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import { useQueryClient } from '@tanstack/react-query';
+import { AppNavHeader } from '../../components/AppNavHeader';
+import { navigationScreenListeners } from '../../hooks/useLogoHeaderOptions';
+import { useDeveloperConfig } from '../../hooks/useDeveloperConfig';
+import {
+  createMessageScreen,
+  MessageTileParams,
+} from '../../screens/MessageScreen';
+import {
+  DirectMessageParams,
+  DirectMessagesScreen,
+} from '../../screens/DirectMessagesScreen';
+import { useAppConfig } from '../../hooks';
+
+type MessageTileStackParamList = {
+  'MessageTileStack/Messages': MessageTileParams;
+  'MessageTileStack/DirectMessage': DirectMessageParams;
+};
+
+const Stack = createNativeStackNavigator<MessageTileStackParamList>();
+
+const MessageScreen = createMessageScreen<MessageTileStackParamList>({
+  DirectMessageScreen: 'MessageTileStack/DirectMessage',
+});
+
+export function MessageTileStack() {
+  const { logoHeaderConfig } = useDeveloperConfig();
+  const { params } = useRoute();
+  const queryClient = useQueryClient();
+  const { data: appConfig } = useAppConfig();
+
+  const tileId = appConfig?.homeTab?.messageTiles?.[0]?.id;
```

Comment:
Do we already, or are you planning, handling what happens if `tileId` is undefined?  This would apply currently and after any break from `homeTab`.

Reply:
Yeah right now it just shows a blank screen. I suppose we could hide the tab if there isn't one configured. Do you think that would be better? Happy to update.

Ultimately I think we need to rethink some of the message tile work.

Reply:
I'll leave it up to you - we have a number of places that have assume-happy-path logic; this definitely isn't the only place.  Another approach is to add some tasks to do some sweeping "increase defensive code & error handling" PRs when there's time.